### PR TITLE
Fix storage save methods to only receive filename

### PR DESCRIPTION
**Pull request by Valleo4**
Storage save method accepted Paths as argument, but was handled as a filename string. This changes the method to only receive a filename to save, then we can add Path functionality later if we need it.

- [x] closes issue #
- [x] Tests added
- [x] Passes flake8
- [x] Updated CHANGELOG
- [ ] Updated README.md

### CODE REVIEW

File: tests/test_artifactory.py [hunk 1]
```diff
@@ -29,18 +29,21 @@ def test_can_load_from_artifactory(open_estimator_pickle):
...
+    open_estimator_pickle, tmp_path: pathlib.Path, regression
+):
     file_path = tmp_path.joinpath("temp.pkl")
 
     def mock_deploy_file(*args, **kwargs):
         return file_path.write_bytes(open_estimator_pickle.read())
 
     mock = MagicMock()
     mock.__truediv__.deploy_file.return_value = mock_deploy_file()
+    mock.is_file.return_value = False
```
Comment (andersbogsnes) [inline tests/test_artifactory.py]: I would add some `mock.assert_called_` statements to check the internals as well

File: tests/test_artifactory.py [hunk 2]
```diff
@@ -29,18 +29,21 @@ def test_can_load_from_artifactory(open_estimator_pickle):
...
+def test_can_save_to_artifactory(
+    open_estimator_pickle, tmp_path: pathlib.Path, regression
+):
     file_path = tmp_path.joinpath("temp.pkl")
 
     def mock_deploy_file(*args, **kwargs):
         return file_path.write_bytes(open_estimator_pickle.read())
 
     mock = MagicMock()
     mock.__truediv__.deploy_file.return_value = mock_deploy_file()
```
Comment (andersbogsnes) [inline tests/test_artifactory.py]: Shouldn't this be `mock.__truediv__.deploy_file  = mock_deploy_file`?

File: src/ml_tooling/storage/artifactory.py [hunk 3]
```diff
@@ -72,14 +72,14 @@ def get_list(self) -> List["ArtifactoryPath"]:
 
         return sorted(self.artifactory_path.glob("*/*.pkl"))
 
-    def load(self, filename: Pathlike) -> Estimator:
+    def load(self, file_path: Pathlike) -> Estimator:
```
Comment (andersbogsnes) [inline src/ml_tooling/storage/artifactory.py]: Does this still work with return values from `storage.list_estimators`? Is there a test for that?

Comment (Valleo4) [reply src/ml_tooling/storage/artifactory.py]: No it didnt and we didnt have a test for it either :) added it in https://github.com/andersbogsnes/ml_tooling/pull/346/commits/56cdb8c8e205423beff6e5ae02ade4c8558fc42d

**Pull request merged**